pkg/kube/controllers/quarkssecret/quarkssecret_controller.go

```diff
@@ -64,12 +64,27 @@ func AddQuarksSecret(ctx context.Context, config *config.Config, mgr manager.Man
 			n := e.ObjectNew.(*qsv1a1.QuarksSecret)
 			o := e.ObjectOld.(*qsv1a1.QuarksSecret)
 
-			// reconcile if it was already generated and the spec changed
+			// reconcile if it was already generated and the spec changed except for `SecretLabels` & `SecretAnnotations`
+			if o.Status.IsGenerated() {
+				for _, key := range []string{"Type", "Request", "SecretName", "Copies"} {
```

Comment:
FWIW, we might want to switch to using [`NumField()`](https://pkg.go.dev/reflect#Value.NumField) + [`FieldByIndex`](https://pkg.go.dev/reflect#Value.FieldByIndex) instead, so that when more fields get added we don't miss them.